uuid/uuid.go

```diff
@@ -1,74 +1,74 @@
-package uefi
+package uuid
 
 // This package implements the mixed-endian UUID as implemented by Microsoft.
 
 import (
 	"encoding/hex"
-	"errors"
+	"fmt"
+	"strings"
 )
 
-var ErrInvalidUUID = errors.New("Invalid UUID")
-
 const (
-	SizeAsBytes  = 16
-	SizeAsString = 36
+	// Size represents number of bytes in a UUID
+	Size = 16
+	// UExample is a example of a string UUID
+	UExample  = "01234567-89AB-CDEF-0123-456789ABCDEF"
+	textLen   = len(UExample)
+	strFormat = "%02X%02X%02X%02X-%02X%02X-%02X%02X-%02X%02X-%02X%02X%02X%02X%02X%02X"
+)
+
+var (
+	fields = [...]int{4, 2, 2, 2, 1, 1, 1, 1, 1, 1}
 )
 
-type UUID struct {
-	Data []byte
+// UUID represents a unique identifier.
+type UUID [Size]byte
```

Comment:
I'm wondering why you can't use github.com/google/uuid. That's what I did in the u-root gpt package. You're doing a lot of work here that the uuid package already does ...

Reply:
that doesn't do endianess. IIRC it assumes big endian

Reply:
that's exactly why I originally reimplemented it instead of using an existing package